proposals/lip-0053.md

```diff
@@ -645,27 +711,53 @@ If a CCM is invalid (as specified in the ["Execute Cross-chain Updates" section]
+When a sidechain is started and registered, the sidechain developers might decide to not activate the sidechain straight away (maybe to do further testing). It could happen then (intentionally or not) that an old block header (almost 30 days old) is submitted to the mainchain to activate the sidechain. This could result in the sidechain being terminated for liveness failure very soon after the activation (maybe only a few minutes later). To prevent this issue (and without any significant drawbacks) the first cross-chain update to be submitted on mainchain must contain a certificate less than 15 days old. The sidechain has therefore at least 15 days to submit the next cross-chain update to the mainchain and start the regular posting of cross-chain updates.
 
 ### Computing the Validators Update
 
-When posting a CCU transaction, the validators hash given in the certificate certifies the new set of validators of the sending chain. The CCU must therefore include the difference between the validators currently stored in the chain account and the validator set authenticated by the certificate. The required `activeValidatorsUpdate` can be obtained by applying the function below.
+When posting a CCU transaction, the validators hash given in the certificate certifies the new set of validators of the sending chain. The CCU must therefore include the difference between the validators currently stored in the chain account and the validator set authenticated by the certificate. The BLS keys of new validators are contained in the `blsKeysUpdate` property, while the new BFT weights are contained in the `bftWeightsUpdate` property. These properties can be obtained applying the function given below.
 
 ```python
-def getActiveValidatorsDiff(currentValidators: list[ActiveValidator], newValidators: list[ActiveValidator]) -> list[ActiveValidator]:
-    activeValidatorsUpdate = []
+def getActiveValidatorsUpdate(currentValidators: list[ActiveValidator], newValidators: list[ActiveValidator]) -> ActiveValidatorsUpdate:
+    # Get the blsKeysUpdate.
+    currentBLSKeys = [v.blsKey for v in currentValidators]
+    newBLSKeys = [v.blsKey for v in newValidators]
+    blsKeysUpdate = sorted([blsKey for blsKey in newBLSKeys if blsKey not in currentBLSKeys])
 
+    validatorsUpdate: list[ActiveValidator] = []
```

Comment:
`validatorsUpdate = []` ?

Reply:
I wanted to be super specific with the type, maybe not necessary? the notation is not wrong

Reply:
ok, we don't use anywhere else. But if you prefer to have it, it's fine.